File: 0204-count-primes/0204-count-primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        if(n<=2){
            return 0;
        }
        vector<bool> isPrime(n+1, true);
    
        isPrime[0]=isPrime[1] = false;
        int count = n-2;

        for (int p = 2; p * p < n; ++p) {
            if (isPrime[p]) {
                for (int i = p * p; i < n; i += p) {
                    if (isPrime[i]) {
                        isPrime[i] = false;
                        count--;
                    }
                }
            }
        }
        return count;
    }
};

File: 0204-count-primes/0204-count-primes.java
class Solution {
    public int countPrimes(int n) {
        if(n<2){
            return 0;
        }
        int count = 0;
        boolean[] arr = new boolean[n];
        for(int i=0;i<n;i++){
            arr[i] = true;
        }

        for(int i=2;i*i<n;i++){
            for(int j=i*i;j<n;j+=i){
                if(j%i==0){
                    arr[j]=false;
                }
            }
        }

        for(boolean b : arr){
            if(b==true){
                count++;
            }
        }

        return count-2;
    }
}